PF/Assignment1/StaxParser.java

```diff
@@ -0,0 +1,190 @@
+import javax.xml.stream.events.EndElement;
+import javax.xml.stream.events.StartElement;
+import javax.xml.stream.events.XMLEvent;
+
+// Class to read data from XML file
+public class StaxParser {
+	static final String VEHICLE = "vehicle";
+	static final String CAR = "car";
+	static final String BIKE = "bike";
+	static final String VEHICLE_ID = "vehicleId";
+	static final String MAKE = "make";
+	static final String MODEL = "model";
+	static final String ENGINE_IN_CC = "engineInCC";
+	static final String FUEL_CAPACITY = "fuelCapacity";
+	static final String MILAGE = "milage";
+	static final String PRICE = "price";
+	static final String ROADTAX = "roadTax";
+	static final String CREATED_BY = "createdBy";
+	static final String AC = "ac";
+	static final String POWER_STEERING = "powerStering";
+	static final String ACCESSORY_KIT = "accessoryKit";
+	static final String SELF_START = "selfStart";
+	static final String HELMET_PRICE = "helmetPrice";
+
+	// -Function to read data from XML
+	public List<Vehicle> readConfig(String configFile) {
+		List<Vehicle> vehicleObjectList = new ArrayList<Vehicle>();
+		try {
+			// First, create a new XMLInputFactory
+			XMLInputFactory inputFactory = XMLInputFactory.newInstance();
+			// Setup a new eventReader
+			InputStream in = new FileInputStream(configFile);
+			XMLEventReader eventReader = inputFactory.createXMLEventReader(in);
+			// read the XML document
+			Vehicle objVehicle = null;
+
+			while (eventReader.hasNext()) {
+				XMLEvent event = eventReader.nextEvent();
+
+				if (event.isStartElement()) {
+					StartElement startElement = event.asStartElement();
+					// If we have an item element, we create a new item
+					if (startElement.getName().getLocalPart() == (CAR)) {
+						objVehicle = new Car();
+					} else if (startElement.getName().getLocalPart() == (BIKE)) {
+						objVehicle = new Bike();
+					}
+
+					if (event.isStartElement()) {
+						if (event.asStartElement().getName().getLocalPart()
+								.equals(VEHICLE_ID)) {
+							event = eventReader.nextEvent();
+							objVehicle.setVehicleId(Integer.parseInt(event
+									.asCharacters().getData()));
+							continue;
+						}
+					}
+
+					if (event.isStartElement()) {
+						if (event.asStartElement().getName().getLocalPart()
+								.equals(MAKE)) {
+							event = eventReader.nextEvent();
+							objVehicle.setMake(event.asCharacters().getData());
+							continue;
+						}
+					}
+
+					if (event.asStartElement().getName().getLocalPart()
+							.equals(MODEL)) {
+						event = eventReader.nextEvent();
+						objVehicle.setModel(event.asCharacters().getData());
+						continue;
+					}
+
+					if (event.asStartElement().getName().getLocalPart()
+							.equals(ENGINE_IN_CC)) {
+						event = eventReader.nextEvent();
+						objVehicle.setEngineInCC(Double.parseDouble(event
+								.asCharacters().getData()));
+						continue;
+					}
+
+					if (event.asStartElement().getName().getLocalPart()
```

Comment:
ACTION: Can't we use else if? we don't need to write continue statement in each if block.